### PR TITLE
Sort providers

### PR DESCRIPTION
**Pull request by ktdreyer**
By sorting the providers alphabetically, we make it less likely that each new in-progress driver will conflict with each other when merging in Git.

DNS was already sorted in #745 , so this PR sorts the `Providers` for the rest.

### CODE REVIEW

**Comment by Kami** [comment]
Thanks.

Please also add a comment to each doc string class otherwise things will get out-of-sync soon :)

Ideally this would also be auto-sorted on commit, but comment is fine to start with.

**Comment by ktdreyer** [comment]
thanks @Kami , I've added another commit to this branch which adds comments above every `Provider` class.

**Comment by ktdreyer** [comment]
Shoot you said to add it to the _doc string_, and I just added it as an inline comment. I will fix that and re-push.

**Comment by ktdreyer** [comment]
I've re-pushed with the doc string comment additions

**Comment by ktdreyer** [comment]
Ok, this passes Travis CI now.

File: libcloud/compute/types.py [hunk 1]
```diff
@@ -59,112 +59,115 @@ class Provider(Type):
...
+    :cvar ABIQUO: Abiquo driver
+    :cvar ALIYUN_ECS: Aliyun ECS driver.
+    :cvar AURORACOMPUTE: Aurora Compute driver.
+    :cvar AZURE: Azure driver.
+    :cvar BLUEBOX: Bluebox
+    :cvar CLOUDSIGMA: CloudSigma
+    :cvar CLOUDSIGMA_US: CloudSigma US Las Vegas
+    :cvar CLOUDSTACK: CloudStack
+    :cvar DIMENSIONDATA: Dimension Data Cloud
+    :cvar EC2_EU_WEST: Amazon AWS EU Ireland
```
Comment (tonybaloney) [inline libcloud/compute/types.py:75]: The compute driver for EC2 is now a single driver. these cvars are wrong @ktdreyer

**Comment by ktdreyer** [comment]
Yeah, there's several inconsistencies in the cvars. I can fix those in a separate PR, if you merge this one?

**Comment by Kami** [comment]
Merged into trunk, thanks.

I've fixed the cvar docstring locally.

**Comment by ktdreyer** [comment]
Thanks @Kami !

**Pull request closed without merging**